pkg/controller/container-runtime-config/helpers.go

```diff
@@ -495,29 +496,64 @@ func validateUserContainerRuntimeConfig(cfg *mcfgv1.ContainerRuntimeConfig) erro
 // getValidBlockedRegistries gets the blocked registries in the image spec and validates that the user is not adding
 // the registry being used by the payload to the list of blocked registries.
 // If the user is, we drop that registry and continue with syncing the registries.conf with the other registry options
-func getValidBlockedRegistries(releaseImage string, imgSpec *apicfgv1.ImageSpec) ([]string, error) {
+func getValidBlockedRegistries(releaseImage string, imgSpec *apicfgv1.ImageSpec, icspRules []*apioperatorsv1alpha1.ImageContentSourcePolicy) ([]string, error) {
```

Comment:
Do we need a similar validation for `allowedRegistries` to ensure that the `payloadRepo` is allowed?

Alternatively, we could hard-code that to be allowed in `policy.json`, as was discussed for `blockedRegistries = ["quay.io"]`; then we wouldn’t quite need to _validate_ `allowedRegistries`.

Reply:
@mtrmac are you saying we should always add the payloadRepo to the allowed list even when the user isn't trying to block it?

Reply:
The first suggestion is essentially a new feature: we now validate `blockedRegistries` to make sure it does not block `payloadRepo`. But if the user sets up very restrictive `allowedRegistries` that does not allow `payloadRepo`, which is, in effect, the same kind of failure, the MCO does not complain.

I no longer think we strictly need that new feature:


The “alternatively” train of thought ultimately leads to https://github.com/openshift/machine-config-operator/pull/2819#discussion_r838767606 : the concept of validating `blockedRegistries` and _completely dropping_ entries that block `payloadRepo` is not quite what we want — we want to specially allow `payloadRepo` _but not more_. If we do that by ~hard-coding that {in the `policy.json` generation code, by submitting appropriate configuration to the `policy.json` generation code}, it seems possible (but I didn’t actually write the code), that the implementation could with very little extra effort handle the `allowedRegistries` configuration as well, making it unnecessary to specifically _validate_ `allowedRegistries`.

Reply:
Taking this into account I have updated the code to allow the payload repo if we detect there is an attempt to block it. PTAL at the update code

Reply:
This works correctly for the validation of `blockedRegistries`.

It doesn’t enforce payload repo presence in `allowedRegistries`, but that would really be a new feature… let’s not increase the scope of this PR again.